pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -163,6 +165,88 @@ func New(ctx context.Context, clients *wrangler.Context) *Planner {
 	}
 }
 
+func (p *Planner) applyToMachineCondition(clusterPlan *plan.Plan, machineNames []string, messagePrefix string, messages map[string][]string) error {
+	var cond condition.Cond
+	var waiting bool
+	for _, machineName := range machineNames {
+		machine := clusterPlan.Machines[machineName]
+		if machine == nil {
+			return fmt.Errorf("found unexpected machine %s that is not in cluster plan", machineName)
+		}
+
+		cond = rke2.Provisioned
+		if rke2.Provisioned.IsTrue(machine) {
+			cond = rke2.Updated
+		}
+
+		machine = machine.DeepCopy()
+		if message := messages[machineName]; len(message) > 0 {
+			msg := strings.Join(message, ", ")
+			waiting = true
+			if cond.GetMessage(machine) == msg {
+				continue
+			}
+			conditions.MarkUnknown(machine, capi.ConditionType(cond), "Waiting", msg)
+		} else if cond.IsTrue(machine) {
+			continue
+		} else {
+			// Even though we are technically not waiting for something, an error should be returned so that the planner will retry.
+			// The machine being updated will cause the planner to re-enqueue with the new data.
+			waiting = true
+			conditions.MarkTrue(machine, capi.ConditionType(cond))
+		}
+
+		if _, err := p.machines.UpdateStatus(machine); err != nil {
+			return err
+		}
+	}
+
+	if waiting {
+		return ErrWaiting(messagePrefix + atMostThree(machineNames) + detailMessage(machineNames, messages))
+	}
+	return nil
+}
+
+func atMostThree(names []string) string {
+	sort.Strings(names)
+	if len(names) > 3 {
+		return strings.Join(names[:3], ",")
```

Comment:
Should we add some type of indicator here to indicate that there are in fact more than 3 `names`? like a `and more` or something

Reply:
Yes I can do that.